depends/packages/zeromq.mk

```diff
@@ -3,9 +3,10 @@ $(package)_version=4.2.1
 $(package)_download_path=https://github.com/zeromq/libzmq/releases/download/v$($(package)_version)/
 $(package)_file_name=$(package)-$($(package)_version).tar.gz
 $(package)_sha256_hash=27d1e82a099228ee85a7ddb2260f40830212402c605a4a10b5e5498a7e0e9d03
+$(package)_dependencies=libsodium
 
 define $(package)_set_vars
-  $(package)_config_opts=--without-documentation --disable-shared --disable-curve
+  $(package)_config_opts=--without-documentation --disable-shared --with-libsodium
```

Comment:
can you explain this change?

Reply:
Removing `--disable-curve` enables CurveMQ, ZeroMQ's transport-layer encryption. It can optionally use libsodium to implement the primitives, which I am choosing to do here given that we already depend on it.

Reply:
why were we disabling CurveMQ before?

Reply:
We inherited that from Bitcoin Core when we pulled in the ZMQ notification API, presumably because they weren't using it (and so we'd likely have set the same options, like we do with OpenSSL, to reduce the code we are depending on).

Reply:
Does this affect compatibility with any clients that were previously using ZMQ without authentication?

Reply:
No - the server has to explicitly enable CurveMQ, and this PR does not do that for the ZMQ notifications API.